azafea/event_processors/endless/metrics/tests_v3/test_request.py

```diff
@@ -69,3 +69,40 @@ def test_request_builder_invalid():
 
     assert ('Metric request is not in the expected format: '
             '(xxsa{ss}ya(aysxmv)a(ayssumv))') in str(excinfo.value)
+
+
+def test_request_builder_wrong_client_clock():
+    from azafea.event_processors.endless.metrics.v3.model import parse_record
+
+    now = datetime.now(tz=timezone.utc)
+    request = GLib.Variant(
+        '(xxsa{ss}ya(aysxmv)a(ayssumv))',
+        (
+            2000000,   # request relative timestamp (2 secs)
+            int(now.timestamp() - 3600 * 1000000000),  # Absolute timestamp, 1 hour late
```

Comment:
Is this right? On my local system this gives:
```
>>> int(now.timestamp() - 3600 * 1000000000)
-3598373029762
```
Maybe you meant `int((now.timestamp() - 3600) * 1000000000)`? I think it would be clearer if you used `now - timedelta(hours=1)`. I'm not familiar with how the timestamps are used in azafea, though.

Reply:
> Maybe you meant `int((now.timestamp() - 3600) * 1000000000)`? I think it would be clearer if you used `now - timedelta(hours=1)`.

Indeed.